### PR TITLE
[Fix] Add typehint for `DarknetBottleneck`

### PR DESCRIPTION
**Pull request by PeterH0323**
## Motivation

Add typehint for `DarknetBottleneck`

## Modification

`mmdetection/mmdet/models/layers/csp_layer.py`

### CODE REVIEW

File: mmdet/models/layers/csp_layer.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ class DarknetBottleneck(BaseModule):
     Args:
         in_channels (int): The input channels of this Module.
         out_channels (int): The output channels of this Module.
-        expansion (int): The kernel size of the convolution. Default: 0.5
+        expansion (float): The kernel size of the convolution. Default: 0.5
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/csp_layer.py:23]: Default: xxx -> Defaults to xxx

File: mmdet/models/layers/csp_layer.py [hunk 2]
```diff
@@ -34,15 +34,16 @@ class DarknetBottleneck(BaseModule):
...
+                 in_channels: int,
+                 out_channels: int,
+                 expansion: float = 0.5,
+                 add_identity: bool = True,
+                 use_depthwise: bool = False,
+                 conv_cfg: OptConfigType = None,
+                 norm_cfg: ConfigType = dict(
+                     type='BN', momentum=0.03, eps=0.001),
+                 act_cfg: ConfigType = dict(type='Swish'),
+                 init_cfg: OptMultiConfig = None):
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/csp_layer.py:46]:
```suggestion
                 init_cfg: OptMultiConfig = None) -> None:
```

File: mmdet/models/layers/csp_layer.py [hunk 3]
```diff
@@ -34,15 +34,16 @@ class DarknetBottleneck(BaseModule):
...
         super().__init__(init_cfg)
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/csp_layer.py:47]:
```suggestion
        super().__init__(init_cfg=init_cfg)
```

File: mmdet/models/layers/csp_layer.py [hunk 4]
```diff
@@ -65,7 +66,7 @@ def __init__(self,
         self.add_identity = \
             add_identity and in_channels == out_channels
 
-    def forward(self, x):
+    def forward(self, x: Tensor) -> Tensor:
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/csp_layer.py:69]: Add docstrings here

File: mmdet/models/layers/csp_layer.py [hunk 5]
```diff
@@ -111,7 +112,7 @@ def __init__(self,
                  norm_cfg: ConfigType = dict(
                      type='BN', momentum=0.03, eps=0.001),
                  act_cfg: ConfigType = dict(type='SiLU'),
-                 init_cfg: OptMultiConfig = None) -> None:
+                 init_cfg: OptMultiConfig = None):
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/csp_layer.py:115]:
```suggestion
                 init_cfg: OptMultiConfig = None) -> None:
```

File: mmdet/models/layers/csp_layer.py [hunk 6]
```diff
@@ -187,7 +188,7 @@ def __init__(self,
                  norm_cfg: ConfigType = dict(
                      type='BN', momentum=0.03, eps=0.001),
                  act_cfg: ConfigType = dict(type='Swish'),
-                 init_cfg: OptMultiConfig = None) -> None:
+                 init_cfg: OptMultiConfig = None):
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/csp_layer.py:191]:
```suggestion
                 init_cfg: OptMultiConfig = None)-> None:
```

File: mmdet/models/layers/csp_layer.py [hunk 7]
```diff
@@ -20,7 +20,8 @@ class DarknetBottleneck(BaseModule):
     Args:
         in_channels (int): The input channels of this Module.
         out_channels (int): The output channels of this Module.
-        expansion (int): The kernel size of the convolution. Default: 0.5
+        expansion (float): The kernel size of the convolution.
+            Defaults to 0.5
         add_identity (bool): Whether to add identity to the out.
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/csp_layer.py:25]: Same, please check the whole PR

File: mmdet/models/layers/csp_layer.py [hunk 8]
```diff
@@ -20,7 +20,8 @@ class DarknetBottleneck(BaseModule):
     Args:
         in_channels (int): The input channels of this Module.
         out_channels (int): The output channels of this Module.
-        expansion (int): The kernel size of the convolution. Default: 0.5
+        expansion (float): The kernel size of the convolution.
+            Defaults to 0.5
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/csp_layer.py:24]:
```suggestion
            Defaults to 0.5.
```

File: mmdet/models/layers/csp_layer.py [hunk 9]
```diff
@@ -20,7 +20,8 @@ class DarknetBottleneck(BaseModule):
...
         add_identity (bool): Whether to add identity to the out.
             Default: True
```
Comment (BIGWangYuDong) [inline mmdet/models/layers/csp_layer.py:26]:
```suggestion
            Defaults to True.
```

**Comment by BIGWangYuDong** [comment]
LGTM, thanks for your contribution! Please change the PR tile to the CodeCamp format! Also, it is suggested to update all CodeCamp tasks in this single PR.

**Comment by PeterH0323** [comment]
Hi @BIGWangYuDong ，我不是 CodeCamp 中 mmdet typehint 的报名成员，我正在为 MMYOLO 提交新算法的开发过程中发现了这个问题，所以提了这个 PR 😄

**Review by BIGWangYuDong — APPROVED** [review]
LGTM

**Pull request merged**